presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFiltersCollector.java

```diff
@@ -79,33 +84,91 @@ public void collectDynamicFilterDomains(Map<DynamicFilterId, Domain> dynamicFilt
+                                        return requireNonNull(columnsMap.get(probeSymbol), () -> format("Missing probe column for %s", probeSymbol));
+                                    },
+                                    Descriptor::getOperator));
                     return Futures.transform(
                             requireNonNull(futures.get(filterId), () -> format("Missing dynamic filter %s", filterId)),
                             // Construct a probe-side predicate by duplicating the resulting domain over the corresponding columns.
                             domain -> TupleDomain.withColumnDomains(
-                                    probeColumns.stream()
+                                    operatorsMap.entrySet().stream()
                                             .collect(toImmutableMap(
-                                                    column -> column,
-                                                    column -> domain))),
+                                                    Map.Entry::getKey,
+                                                    entry -> applyComparison(entry.getValue(), domain)))),
                             directExecutor());
                 })
                 .collect(toImmutableList());
         return new TableSpecificDynamicFilter(predicateFutures);
     }
 
+    private static Domain applyComparison(ComparisonExpression.Operator operator, Domain domain)
+    {
+        switch (operator) {
+            case EQUAL:
+                return Domain.create(domain.getValues(), false);
+            case LESS_THAN: {
+                List<Range> ranges = domain.getValues().getRanges().getOrderedRanges();
+                Range result = lessThanRange(ranges.get(ranges.size() - 1).getHigh());
+                return Domain.create(ValueSet.ofRanges(result), false);
+            }
+            case LESS_THAN_OR_EQUAL: {
+                List<Range> ranges = domain.getValues().getRanges().getOrderedRanges();
+                Range result = lessThanOrEqualRange(ranges.get(ranges.size() - 1).getHigh());
+                return Domain.create(ValueSet.ofRanges(result), false);
+            }
+            case GREATER_THAN: {
+                List<Range> ranges = domain.getValues().getRanges().getOrderedRanges();
+                Range result = greaterThanRange(ranges.get(0).getLow());
+                return Domain.create(ValueSet.ofRanges(result), false);
+            }
+            case GREATER_THAN_OR_EQUAL: {
+                List<Range> ranges = domain.getValues().getRanges().getOrderedRanges();
+                Range result = greaterThanOrEqualRange(ranges.get(0).getLow());
+                return Domain.create(ValueSet.ofRanges(result), false);
+            }
+            default:
+                throw new IllegalArgumentException("Unsupported dynamic filtering comparison operator: " + operator);
+        }
+    }
+
+    private static Range lessThanRange(Marker high)
```

Comment:
Could we use `Marker#getValue` to get Object and then just reuse the existing functions in `Range.java` ?